range_pool.go

```diff
@@ -0,0 +1,354 @@
+package rangepool
+
+import (
+	"context"
+	"fmt"
+	"sort"
+	"strconv"
+
+	microerror "github.com/giantswarm/microkit/error"
+	micrologger "github.com/giantswarm/microkit/logger"
+	microstorage "github.com/giantswarm/microkit/storage"
+)
+
+const (
+	// IDKeyFormat is the format string used to create a storage key to persist
+	// the relationship between IDs and items.
+	//
+	//     range-pool/${namespace1}/id/${id1}/item/${item1}    ${item1}
+	//     range-pool/${namespace1}/id/${id1}/item/${item2}    ${item2}
+	//     range-pool/${namespace1}/id/${id2}/item/${item3}    ${item3}
+	//     range-pool/${namespace1}/id/${id2}/item/${item4}    ${item4}
+	//
+	IDKeyFormat = "range-pool/%s/id/%s/item/%s"
+	// IDListKeyFormat is the format string used to create a storage key to lookup
+	// the list of items of an ID. See also IDKeyFormat.
+	IDListKeyFormat = "range-pool/%s/id/%s/item"
+	// ItemKeyFormat is the format string used to create a storage key to persist
+	// the relation between a namespace and its associated items.
+	//
+	//     range-pool/${namespace1}/item/${item1}    ${item1}
+	//     range-pool/${namespace1}/item/${item2}    ${item2}
+	//     range-pool/${namespace1}/item/${item3}    ${item3}
+	//     range-pool/${namespace1}/item/${item4}    ${item4}
+	//
+	ItemKeyFormat = "range-pool/%s/item/%s"
+	// ItemListKeyFormat is the format string used to create a storage key to
+	// lookup the list of items of a namespace. See also ItemKeyFormat.
+	ItemListKeyFormat = "range-pool/%s/item"
+	// LatestKeyFormat is used to create a storage key to persist the latest item
+	// used.
+	//
+	//     range-pool/${namespace1}/latest    ${item4}
+	//
+	LatestKeyFormat = "range-pool/%s/latest"
+)
+
+const (
+	// latestItemException indicates there was no latest range pool item, which
+	// means that there has no item ever been before. In this case the range pool
```

Comment:
Nitpick but "which means there has never been an item before." reads better to me.